File: api-gateway/models/Vehicle.js
const mongoose = require('mongoose');

const VehicleSchema = new mongoose.Schema({
    title: { type: String, required: true, maxlength: 255 },
    brand: { type: String, required: true, maxlength: 100 },
    model: { type: String, required: true, maxlength: 100 },
    year: { type: Number, required: true },
    price: { type: Number, required: true },
    mileage: { type: Number, required: true },
    fuelType: { type: String, required: true, maxlength: 50 },
    transmission: { type: String, required: true, maxlength: 50 },
    condition: { type: String, required: true, maxlength: 50 },
    district: { type: String, required: true, maxlength: 100 },
    city: { type: String, required: true, maxlength: 100 },
    description: { type: String, required: false },
    availabilityStatus: { type: String, default: 'Available', maxlength: 20 },
    sellerId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
    videoUrl: { type: String, required: false, maxlength: 255 },
    createdAt: { type: Date, default: Date.now },
    updatedAt: { type: Date, default: Date.now }
}, {
    timestamps: true
});

VehicleSchema.virtual('VehicleImages', {
    ref: 'VehicleImage',
    localField: '_id',
    foreignField: 'vehicleId'
});

VehicleSchema.set('toJSON', { virtuals: true });
VehicleSchema.set('toObject', { virtuals: true });

module.exports = mongoose.model('Vehicle', VehicleSchema);

File: filtering-service/src/main/java/com/example/cartradingsystem/model/Vehicle.java
package com.example.cartradingsystem.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Document(collection = "vehicles")
@Data
public class Vehicle {

    @Id
    private String id;

    private String title;
    private String brand;
    private String model;
    private Integer year;
    private BigDecimal price;
    private Integer mileage;

    @Field("fuelType")
    private String fuelType;

    @Field("transmission")
    private String transmission;

    @Field("condition")
    private String condition;

    private String district;
    private String city;
    private String description;

    @Field("availabilityStatus")
    private String availabilityStatus;

    @Field("sellerId")
    private String sellerId;

    @Field("videoUrl")
    private String videoUrl;

    @Field("createdAt")
    private LocalDateTime createdAt;

    @Field("updatedAt")
    private LocalDateTime updatedAt;
}
